File: CS/NewProject/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.DataAccess;
using DevExpress.XtraEditors;
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;

namespace NewProject
{
    public partial class Form1 : XtraForm
    {
        List<Data> ds = Data.CreateData();
        public Form1()
        {
            InitializeComponent();
            dashboardViewer1.DataLoading += dashboardViewer1_DataLoading;
            dashboardViewer1.DataSourceOptions.ObjectDataSourceLoadingBehavior = DocumentLoadingBehavior.LoadAsIs;
            dashboardViewer1.DashboardSource = typeof(Dashboard1);
        }

        private void dashboardViewer1_DataLoading(object sender, DevExpress.DashboardCommon.DataLoadingEventArgs e) {
            e.Data = ds;
        }

        private void simpleButton1_Click(object sender, EventArgs e)
        {
           DashboardItemCollection controls = dashboardViewer1.Dashboard.Items;
            PrintControlInfo(controls);
        }
        private static void PrintControlInfo(IEnumerable<DashboardItem> controls)
        {
            StringBuilder stringBuilder = new StringBuilder();
            foreach (DashboardItem control in controls)
                stringBuilder.AppendLine(control.GetType().ToString());
            XtraMessageBox.Show(stringBuilder.ToString());
        }
    }
}

File: VB/NewProject/Form1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DataAccess
Imports DevExpress.XtraEditors
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Windows.Forms

Namespace NewProject
	Partial Public Class Form1
		Inherits XtraForm

		Private ds As List(Of Data) = Data.CreateData()
		Public Sub New()
			InitializeComponent()
			AddHandler dashboardViewer1.DataLoading, AddressOf dashboardViewer1_DataLoading
			dashboardViewer1.DataSourceOptions.ObjectDataSourceLoadingBehavior = DocumentLoadingBehavior.LoadAsIs
			dashboardViewer1.DashboardSource = GetType(Dashboard1)
		End Sub

		Private Sub dashboardViewer1_DataLoading(ByVal sender As Object, ByVal e As DevExpress.DashboardCommon.DataLoadingEventArgs)
			e.Data = ds
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
'INSTANT VB NOTE: The variable controls was renamed since Visual Basic does not handle local variables named the same as class members well:
		   Dim controls_Renamed As DashboardItemCollection = dashboardViewer1.Dashboard.Items
			PrintControlInfo(controls_Renamed)
		End Sub
		Private Shared Sub PrintControlInfo(ByVal controls As IEnumerable(Of DashboardItem))
			Dim stringBuilder As New StringBuilder()
			For Each control As DashboardItem In controls
				stringBuilder.AppendLine(control.GetType().ToString())
			Next control
			XtraMessageBox.Show(stringBuilder.ToString())
		End Sub
	End Class
End Namespace